Sources/PointFree/Stripe.swift

```diff
@@ -213,18 +228,40 @@ private func createSubscription(customer: Stripe.Customer.Id, plan: Stripe.Plan.
 }
 
 private func fetchCustomer(id: Stripe.Customer.Id) -> EitherIO<Prelude.Unit, Stripe.Customer> {
-  return stripeDataTask("customers/\(id.unwrap)")
+  return stripeDataTask("customers/" + id.unwrap)
 }
 
 private let fetchPlans: EitherIO<Prelude.Unit, Stripe.ListEnvelope<Stripe.Plan>> =
   stripeDataTask("plans")
 
 private func fetchPlan(id: Stripe.Plan.Id) -> EitherIO<Prelude.Unit, Stripe.Plan> {
-  return stripeDataTask("plans/\(id.rawValue)")
+  return stripeDataTask("plans/" + id.rawValue)
 }
 
 private func fetchSubscription(id: Stripe.Subscription.Id) -> EitherIO<Prelude.Unit, Stripe.Subscription> {
-  return stripeDataTask("subscriptions/\(id.unwrap)?expand[]=customer")
+  return stripeDataTask("subscriptions/" + id.unwrap + "?expand[]=customer")
+}
+
+private func reactivateSubscription(_ subscription: Stripe.Subscription)
+  -> EitherIO<Prelude.Unit, Stripe.Subscription> {
+
+    guard
+      subscription.cancelAtPeriodEnd,
+      let item = subscription.items.data.first
+      else { return throwE(unit) }
+
+    return updateSubscription(subscription, item.plan.id, item.quantity)
+}
+
+private func updateSubscription(_ subscription: Stripe.Subscription, _ plan: Stripe.Plan.Id, _ quantity: Int)
```

Comment:
Thinking we may wanna move back to passing values instead of ids. We never manually build ids (outside of tests), so we have the full values at our disposal. In the case of subscription updates, we need quite a bit of data from the subscription structure:

- Subscription id
- Subscription item id
- Subscription item plan (for reactivation)
- Subscription item quantity (for reactivation)

Reply:
gotcha gotcha. I forget why I started doing that, but we can try it and see what happens